webrob/neems/neem.py

```diff
@@ -18,27 +25,35 @@ def __init__(self,
             b_id = bson.objectid.ObjectId(neem_id)
         else:
             b_id = neem_id
-        neem = mongoDBMetaCollection.find_one({"_id": b_id})
-        self.neem_id = str(neem['_id'])
-        # TODO: Tag could be useful for versioning
-        self.neem_tag = ''
-        self.name = neem['name']
-        self.description = neem['description']
-        self.created_by = neem['created_by']
-        self.created_at = parser.parse(neem['created_at']).strftime('%m/%d/%y %H:%M')
-        self.model_version = neem['model_version']
-        self.downloadUrl = NEEM_DOWNLOAD_URL_PREFIX + neem['url']
-        self.neem_repo_path = neem['url']
-        self.knowrob_image = 'knowrob'
-        self.knowrob_tag = 'latest'
-        self.maintainer = neem['created_by']
-        self.authors = neem['created_by']
-        self.acknowledgements = ''
-        self.environment = neem['environment']
-        self.activity = neem['activity']
-        self.agent = neem['agent']
-        self.keywords = neem['keywords']
-        self.image = neem['image']
+
+        try:
+            mongoDBMetaCollection = get_mongo_db_meta_collection()
+            neem = mongoDBMetaCollection.find_one({"_id": b_id})
+            self.neem_id = str(neem['_id'])
+            # TODO: Tag could be useful for versioning
+            self.neem_tag = ''
+            self.name = neem['name']
+            self.description = neem['description']
+            self.created_by = neem['created_by']
+            self.created_at = parser.parse(neem['created_at']).strftime('%m/%d/%y %H:%M')
+            self.model_version = neem['model_version']
+            self.downloadUrl = NEEM_DOWNLOAD_URL_PREFIX + neem['url']
+            self.neem_repo_path = neem['url']
+            self.knowrob_image = 'knowrob'
+            self.knowrob_tag = 'latest'
+            self.maintainer = neem['created_by']
+            self.authors = neem['created_by']
+            self.acknowledgements = ''
+            self.environment = neem['environment']
+            self.activity = neem['activity']
+            self.agent = neem['agent']
+            self.keywords = neem['keywords']
+            self.image = neem['image']
+        except ConnectionFailure as e:
+            raise NEEMMetaException('An exception has occurred during connection with mongodb collection, please check!', exc=e)
+
+        except PyMongoError as e:
```

Comment:
why not adding a flask handler for this error?

Reply:
done. Now PyMongoError has its own flask handler.